src/lib.rs

```diff
@@ -400,22 +410,52 @@ pub fn encode_to_fmt<T: AsRef<[u5]>>(
     }
 }
 
+/// Used for encode/decode operations for the two variants of Bech32
+#[derive(Copy, Clone, PartialEq, Eq, PartialOrd, Ord, Hash, Debug)]
+pub enum Variant {
```

Comment:
I suggest also to add `repr(u32)` here

Reply:
What does this do for us?

Reply:
It's necessary to implement the `Variant::Bech32 as u32` API afaik (making the enum castable to its internal representation). I don't really like it because it isn't too well-known this is possible and also seems a bit like abusing this feature here for probably non-existent performance gains.

Reply:
Makes sense. I'll leave this out for now, and we can circle back later if the need arises.